### PR TITLE
simavr: init at 1.3

### PR DESCRIPTION
**Pull request by goodrone**
Add simavr

### CODE REVIEW

File: pkgs/development/tools/simavr/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ makeWrapper which git avrgcclibc libelf pkgconfig freeglut mesa ];
+
+  sourceRoot = ".";
+
+  meta = with stdenv.lib; {
+    description = "A lean and mean Atmel AVR simulator for Linux";
+    homepage    = https://github.com/buserror/simavr;
+    license     = licenses.gpl3;
+    maintainers = with maintainers; [ offline ];
```
Comment (bjornfor) [inline pkgs/development/tools/simavr/default.nix]: Is "offline" the maintainer? (Copy paste leftover?)

File: pkgs/development/tools/simavr/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  unpackCmd = ''
+    tar -xvf $src
+    cd simavr-$version
+  '';
+
+  buildFlags = "LINUX_AVR_ROOTS=${avrgcclibc} SIMAVR_VERSION=${version}";
+  installFlags = buildFlags + " DESTDIR=$(out)";
+
+  phases = [ "unpackPhase" "buildPhase" "installPhase" ];
```
Comment (bjornfor) [inline pkgs/development/tools/simavr/default.nix]: FYI, specifying phases are generally not needed.

**Comment by bjornfor** [comment]
Cool, a new simulator! I was just looking at simulavr the other day; I have a half working Nix package file for it.

**Comment by goodrone** [comment]
@bjornfor thank you!  I hope I've addressed these issues.

**Comment by FRidh** [comment]
@goodrone could you squash the commits except for d7ae272835ba733686d0ba1230dcc2cece994ebb

**Comment by goodrone** [comment]
@FRidh sure. I'm currently seeing if I can also get it working for Darwin. Thank you!

File: pkgs/development/tools/simavr/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, makeWrapper, avrgcclibc, libelf, which, git, pkgconfig, freeglut, mesa }:
+
+stdenv.mkDerivation rec {
+  name = "simavr-${version}";
+  version = "1.3";
+  enableParallelBuilding = true;
+
+  libPath = stdenv.lib.makeLibraryPath [
+    libelf
```
Comment (the-kenny) [inline pkgs/development/tools/simavr/default.nix]: Are you sure this is necessary? The build process should actually compile the correct path to libelf directly into the binary.

File: pkgs/development/tools/simavr/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+
+  buildFlags = "AVR_ROOT=${avrgcclibc}/avr SIMAVR_VERSION=${version}";
+  installFlags = buildFlags + " DESTDIR=$(out)";
+
+  postFixup = ''
+    patchelf --set-rpath "$libPath:$out/lib" "$out/bin/simavr"
+  '';
+
+  buildInputs = [ makeWrapper which git avrgcclibc libelf pkgconfig freeglut mesa ];
```
Comment (the-kenny) [inline pkgs/development/tools/simavr/default.nix]: `makeWrapper` seems unused

File: pkgs/development/tools/simavr/default.nix (hunk 3 above)
Comment (goodrone) [inline pkgs/development/tools/simavr/default.nix]: Without libelf, I get:

    $ result/bin/simavr
    result/bin/simavr: error while loading shared libraries: libelf.so.0: cannot open shared object file: No such file or directory

File: pkgs/development/tools/simavr/default.nix (hunk 4 above)
Comment (goodrone) [inline pkgs/development/tools/simavr/default.nix]: @the-kenny  thank you, I'm dropping this dependency.

**Pull request closed without merging**